internal/controller/controller.go

```diff
@@ -234,15 +242,56 @@ func (c *httpController) exportRecordedData(writer http.ResponseWriter, request
 		return
 	}
 
-	jsonResponse, err := json.Marshal(recordedData)
-	if err != nil {
+	if reflect.DeepEqual(recordedData, &dtos.RecordedData{}) {
+		writer.WriteHeader(http.StatusNoContent)
+		_, _ = writer.Write([]byte("no recorded data found"))
+		return
+	}
+
+	compression := request.URL.Query().Get("Content-Encoding")
```

Comment:
I was thinking the query parameter would be `compression`
i.e.
```
/export?compression=GZIP
```
Why did you pick `Content-Encoding` for the query parameter name?
I think the response should sent `Content-Encoding` in the header like you have done. :-)